contracts/FeralfileArtworkV4.sol

```diff
@@ -10,60 +10,216 @@ import "./Authorizable.sol";
+            seriesMaxSupplies_.length > 0,
+            "FeralfileExhibitionV4: _seriesMaxSupplies is empty"
+        );
+        require(
+            seriesIds_.length == seriesMaxSupplies_.length,
+            "FeralfileExhibitionV4: seriesMaxSupplies_ and seriesIds_ lengths are not the same"
+        );
+
+        _signer = signer_;
+        _burnable = burnable_;
+        _bridgeable = bridgeable_;
+
+        // initialize max supply map
+        for (uint256 i = 0; i < seriesIds_.length; i++) {
+            _seriesMaxSupplies[seriesIds_[i]] = seriesMaxSupplies_[i];
+            require(
+                seriesMaxSupplies_[i] > 0,
+                "FeralfileExhibitionV4: zero max supply"
+            );
+        }
     }
 
     function supportsInterface(
         bytes4 interfaceId
-    ) public view virtual override(ERC721Enumerable, IERC165) returns (bool) {
-        return
-            interfaceId == type(IERC721Enumerable).interfaceId ||
-            super.supportsInterface(interfaceId);
+    ) public view virtual override(ERC721) returns (bool) {
+        return super.supportsInterface(interfaceId);
     }
 
-    /// @notice Return a count of series registered in this exhibition
-    function totalSeries() public view virtual returns (uint256) {}
+    function mintable() public view virtual returns (bool) {
+        return _mintable;
+    }
+
+    function setMintable(bool mintable_) external virtual onlyAuthorized {
+        _mintable = mintable_;
+    }
+
+    function burnable() public view virtual returns (bool) {
+        return _burnable;
+    }
+
+    function setBurnable(bool burnable_) external virtual onlyAuthorized {
+        _burnable = burnable_;
+    }
+
+    function bridgeable() public view virtual returns (bool) {
+        return _bridgeable;
+    }
+
+    function setBridgeable(bool bridgeable_) external virtual onlyAuthorized {
+        _bridgeable = bridgeable_;
+    }
+
+    function signer() public view virtual returns (address) {
+        return _signer;
+    }
+
+    function totalSupply() public view virtual returns (uint256) {
+        return _totalSupply;
+    }
+
+    function seriesMaxSupply(
+        uint256 seriesId
+    ) public view virtual returns (uint256) {
+        return _seriesMaxSupplies[seriesId];
+    }
+
+    function seriesTotalSupply(
+        uint256 seriesId
+    ) public view virtual returns (uint256) {
+        return _seriesTotalSupplies[seriesId];
+    }
 
-    /// @notice Update the IPFS cid of an artwork to a new value
-    function updateArtworkIPFSCid(
+    function getArtwork(
+        uint256 tokenId
+    ) public view virtual returns (Artwork memory) {
+        require(_exists(tokenId), "ERC721: token doesn't exist");
+        return _allArtworks[tokenId];
+    }
+
+    function setApprovalForAll(
+        address operator,
+        bool approved
+    ) public override(ERC721) onlyAllowedOperatorApproval(operator) {
+        super.setApprovalForAll(operator, approved);
+    }
```

Comment:
remove this. we no need to override cause you don't add any customized logic.
and ERC721 is the only contract that implement the `setApprovalForAll`

Reply:
this is for `operatorRegistry` so it is required.

Reply:
Oh I see, I'll remove the related comments